File: internal/pkg/dto/response/conversation_copilot.go
package response

type CopilotKnowledgeSource struct {
	KnowledgeBaseID int64  `json:"knowledgeBaseId"`
	DocumentID      int64  `json:"documentId"`
	ChunkID         int64  `json:"chunkId"`
	Title           string `json:"title"`
	Content         string `json:"content"`
}

type CopilotSkillTrace struct {
	ID     int64  `json:"id"`
	Name   string `json:"name"`
	Reason string `json:"reason"`
}

type CopilotToolTrace struct {
	Code   string `json:"code"`
	Status string `json:"status"`
}

type ConversationReplySuggestion struct {
	HandoffRequested bool                     `json:"handoffRequested,omitempty"`
	HandoffReason    string                   `json:"handoffReason,omitempty"`
	ConversationID   int64                    `json:"conversationId"`
	LastMessageID    int64                    `json:"lastMessageId"`
	AgentName        string                   `json:"agentName"`
	ModelName        string                   `json:"modelName"`
	Content          string                   `json:"content"`
	SkillStatus      string                   `json:"skillStatus"`
	Skills           []CopilotSkillTrace      `json:"skills"`
	Tools            []CopilotToolTrace       `json:"tools"`
	KnowledgeStatus  string                   `json:"knowledgeStatus"`
	Sources          []CopilotKnowledgeSource `json:"sources"`
	RoutingMs        int64                    `json:"routingMs"`
	RetrievalMs      int64                    `json:"retrievalMs"`
	GenerationMs     int64                    `json:"generationMs"`
	DurationMs       int64                    `json:"durationMs"`
}

File: web/lib/api/conversation-copilot.ts
import { request } from "@/lib/api/client";

export type ReplySuggestion = {
  conversationId: number;
  lastMessageId: number;
  agentName: string;
  modelName: string;
  content: string;
  skillStatus: "matched" | "none" | "failed" | "not_configured";
  skills: { id: number; name: string; reason: string }[];
  tools: { code: string; status: "used" | "matched" | "empty" }[];
  knowledgeStatus: "matched" | "empty" | "not_configured";
  sources: { knowledgeBaseId: number; documentId: number; chunkId: number; title: string; content: string }[];
  routingMs: number;
  retrievalMs: number;
  generationMs: number;
  durationMs: number;
};

export function suggestConversationReply(id: number, signal?: AbortSignal) {
  return request<ReplySuggestion>(`/api/dashboard/conversation/${id}/suggest_reply`, { method: "POST", signal });
}
